NitroxClient/GameLogic/Vehicles.cs

```diff
@@ -163,28 +213,44 @@ private void OnVehiclePrefabLoaded(TechType techType, GameObject prefab, NitroxI
                 {
                     vehicle.vehicleName = name;
                     vehicle.subName.DeserializeName(vehicle.vehicleName);
+
+                    //Log.Debug($"Set vehicle name to {name} for {id}");
                 }
 
                 if (colours != null)
                 {
+                    //Log.Debug($"Updating vehicle colors for {id}");
+
                     Vector3[] colour = new Vector3[5];
 
                     for (int i = 0; i < hsb.Length; i++)
                     {
                         colour[i] = hsb[i];
                     }
+                    
                     vehicle.vehicleColors = colour;
                     vehicle.subName.DeserializeColors(vehicle.vehicleColors);
+
+                    //Log.Debug($"Updated vehicle colors for {id}");
                 }
 
                 vehicle.GetComponent<LiveMixin>().health = health;
             }
             else if(techType == TechType.Cyclops)
             {
                 GameObject target = NitroxEntity.RequireObjectFrom(id);
+                if (target == null)
```

Comment:
Any object that inherits from `UnityEngine.Object` should not be compared against `== null` but instead just do `if (obj)` so that the implicit bool operator that Unity defines for these objects is triggered.

This is required because Unity objects can be non-null but dead. Which isn't good either.